src/OrbitGl/include/OrbitGl/CallTreeView.h

```diff
@@ -181,27 +181,60 @@ class CallTreeUnwindErrorType : public CallTreeNode {
       : CallTreeNode{parent}, error_type_{error_type} {
     ORBIT_CHECK(error_type != orbit_client_data::CallstackType::kComplete);
   }
+  ~CallTreeUnwindErrorType() override = default;
 
   [[nodiscard]] orbit_client_data::CallstackType error_type() const { return error_type_; }
 
  private:
   orbit_client_data::CallstackType error_type_;
 };
 
-class CallTreeView : public CallTreeNode {
+class CallTreeRoot : public CallTreeNode {
+ public:
+  CallTreeRoot() : CallTreeNode(nullptr) {}
+  ~CallTreeRoot() override = default;
+};
+
+class CallTreeView {
  public:
   [[nodiscard]] static std::unique_ptr<CallTreeView> CreateTopDownViewFromPostProcessedSamplingData(
       const orbit_client_data::PostProcessedSamplingData& post_processed_sampling_data,
-      const orbit_client_data::ModuleManager& module_manager,
-      const orbit_client_data::CaptureData& capture_data);
+      const orbit_client_data::ModuleManager* module_manager,
+      const orbit_client_data::CaptureData* capture_data);
 
   [[nodiscard]] static std::unique_ptr<CallTreeView>
   CreateBottomUpViewFromPostProcessedSamplingData(
       const orbit_client_data::PostProcessedSamplingData& post_processed_sampling_data,
-      const orbit_client_data::ModuleManager& module_manager,
-      const orbit_client_data::CaptureData& capture_data);
+      const orbit_client_data::ModuleManager* module_manager,
+      const orbit_client_data::CaptureData* capture_data);
+
+  CallTreeView(std::unique_ptr<CallTreeRoot> call_tree_root,
```

Comment:
I think this constructor should be private as you have factory functions. `make_unique` won't work, but you can use `new` and `absl::WrapUnique`.

Reply:
Done.